core/src/test/scala/unit/kafka/server/ReplicaFetcherThreadTest.scala

```diff
@@ -130,96 +128,12 @@ class ReplicaFetcherThreadTest {
       ApiKeys.FETCH.latestVersion(true),
       testingVersion.fetchRequestVersion
     )
-    assertEquals(
-      ApiKeys.OFFSET_FOR_LEADER_EPOCH.latestVersion(true),
-      testingVersion.offsetForLeaderEpochRequestVersion
-    )
     assertEquals(
       ApiKeys.LIST_OFFSETS.latestVersion(true),
       testingVersion.listOffsetRequestVersion
     )
   }
 
-  @Disabled("KAFKA-18370")
-  @Test
-  def testFetchLeaderEpochRequestIfLastEpochDefinedForSomePartitions(): Unit = {
```

Comment:
Lots of disabled tests that rely on the "no truncate on fetch", which is no longer applicable.

Reply:
Sorry @junrao - I didn't fully understand. Are you saying we should _not_ remove these tests just yet or that it's ok to remove these and remove the rest in a separate PR?

Reply:
KAFKA-18370 says that we could remove these tests because 4.0 doesn't support ZK anymore. This seems incorrect since this part of the code is independent of the ZK change. However, it does seem that the OffsetForLeaderEpochRequest is not really exercised because of the truncating on fetch support. I am just not 100% sure. So, it would useful to try to remove that part of the code first and then remove the corresponding tests.

cc @chia7712

Reply:
I see. The reason I removed these tests is:

1. They relied on IBP 2.6-IV0 via the `kafkaConfigNoTruncateOnFetch` method.
2. They were disabled.
3. I was under the impression that the truncation approach from IBP 2.7 onwards does not rely on offset for leader epochs requests: #9382

So, I wasn't sure how to make the tests work. 

Your point is that the legacy code is still there and hence removing the tests is premature. Is that right?

Reply:
Since these tests were already disabled, is it OK if I keep them deleted and follow up with a separate change that removes the supposed replica fetcher dead code?

Reply:
Sounds good. I filed https://issues.apache.org/jira/browse/KAFKA-18477 to track the removal work in replica fetcher.